### DISCUSSION TITLE
Disallow custom value in Combobox

### DISCUSSION THREAD

**Discussion by ibrahimbutt** [ariakit/ariakit#3254] [Questions] [answered]
I require a combobox which does not allow a custom value. In essence, a select with filtering and focus on the first match.

How can I achieve this with the combobox?

There is an example using Select with a Combobox inside, but I don't want the input to be inside the popover.

Besides an example, could we have this feature through a prop?

I've read the W3 guideline on combobox, and they can have either functionality.

**Comment by diegohaz** [comment, marked as answer]
Hi @ibrahimbutt!

The [Select with Combobox](https://ariakit.org/examples/select-combobox) pattern is really the best choice for that scenario as it clearly communicates to users that they must pick a value from the given list. With a free-text input, there's no such expectation, there's a chance users might get frustrated when they discover the value they typed in can't be used.

If you still don't want to use Select, the specific implementation will differ based on your use case. If "foo" is a valid option and the user types "foo" without selecting it from the list, should the value be kept? Or can they only pick the values from the list (by clicking, pressing enter, etc.)? What if "Apple" is a valid value, and the user types "apple" (with lowercase "a")?

Also, could you please link to the W3C guidelines that discuss this?

**Reply by ibrahimbutt** [reply]
After some thought, I also believe it would be better to go with the select and combo aproach. Your comment reinforces this!

The guidlines can be found [here]( https://www.w3.org/WAI/ARIA/apg/patterns/combobox/), with [this](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/#:~:text=Combobox%20widgets%20are,the%20user%20time) section being what we are interested in.

**Reply by diegohaz** [reply]
Oh, thanks! Note that the combobox pattern on APG also includes select widgets, which are technically select-only comboboxes.
